onmt/IO.py

```diff
@@ -267,67 +305,78 @@ class ONMTDataset(torchtext.data.Dataset):
     """
     Defines a dataset for machine translation.
     An ONMTDataset is a collection that supports iteration over its
-    examples. The parent class supports indexing as well, but future
-    developments here may make that difficult (lazy iteration over
-    examples because of large datasets, for example).
+    examples.
     """
 
     @staticmethod
     def sort_key(ex):
         "Sort in reverse size order"
         return -len(ex.src)
 
-    def __init__(self, src_path, tgt_path, fields,
+    @staticmethod
+    def coalesce_datasets(datasets):
+        """Coalesce all ONMTDataset instances."""
+        final = datasets[0]
+        for d in datasets[1:]:
+            final.src_vocabs += d.src_vocabs
+
+            # All datasets have same number of features.
+            aeq(final.n_src_feats, d.n_src_feats)
+            aeq(final.n_tgt_feats, d.n_tgt_feats)
+
+            final.examples += d.examples
```

Comment:
Is this lazy? or are you actually loading in all of the datasets at training time. Won't that cause the same memory issue>